### PR TITLE
refactor: Refactored and reformatted for better readability and efficiency

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
# PR Description
I used [`ruff`](https://docs.astral.sh/ruff/), [`Pylint`](https://pylint.readthedocs.io/en/stable/) linters to identify and fix some of the changes.

## Related Issue
Closes #

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### Socials
https://twitter.com/Sai_Suraj_27

### CODE REVIEW

File: ivy/functional/backends/numpy/random.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def randint(
 
 def seed(*, seed_value: int = 0) -> None:
```
Comment (KareemMAX) [inline ivy/functional/backends/numpy/random.py:114]: I think this should be a void function?

File: ivy/functional/backends/paddle/experimental/elementwise.py [hunk 2]
```diff
@@ -751,9 +751,11 @@ def _is_scalar(x):
     """Determines if the given tensor is a scalar.
 
     Args:
```
Comment (KareemMAX) [inline ivy/functional/backends/paddle/experimental/elementwise.py:753]:
```suggestion
    Args
```

File: ivy/functional/backends/paddle/experimental/elementwise.py [hunk 3]
```diff
@@ -751,9 +751,11 @@ def _is_scalar(x):
...
+    ----
     - x (paddle.Tensor): Input tensor.
 
     Returns:
```
Comment (KareemMAX) [inline ivy/functional/backends/paddle/experimental/elementwise.py:757]:
```suggestion
    Returns
```

File: ivy/functional/backends/paddle/random.py [hunk 4]
```diff
@@ -142,7 +142,7 @@ def randint(
 
 def seed(*, seed_value: int = 0) -> None:
```
Comment (KareemMAX) [inline ivy/functional/backends/paddle/random.py:143]: I think this should be a void function? Maybe @vedpatwardhan can chime in here as looks like this is a systematic error (which I'm not sure what are the consequences of removing it)

File: ivy/functional/backends/tensorflow/random.py [hunk 5]
```diff
@@ -146,7 +146,7 @@ def randint(
 
 def seed(*, seed_value: int = 0) -> None:
```
Comment (KareemMAX) [inline ivy/functional/backends/tensorflow/random.py:147]: Same issue with void function here @vedpatwardhan

File: ivy/functional/backends/torch/random.py [hunk 6]
```diff
@@ -124,7 +124,7 @@ def seed(*, seed_value: int = 0) -> None:
             from torch import mps
 
             mps.manual_seed(seed_value)
-    return
+    return None
```
Comment (KareemMAX) [inline ivy/functional/backends/torch/random.py:127]: Same issue with void function here @vedpatwardhan

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 7]
```diff
@@ -329,7 +329,7 @@ def flatten(self, order="C"):
 
     def fill(self, num, /):
         self.ivy_array = np_frontend.full(self.shape, num).ivy_array
-        return
+        return None
```
Comment (KareemMAX) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:332]: Same issue with the void function @vedpatwardhan

File: ivy/functional/frontends/torch/dtype.py [hunk 8]
```diff
@@ -44,4 +44,4 @@ def set_default_dtype(d):
     )
     global _default_dtype
     _default_dtype = d
-    return
+    return None
```
Comment (KareemMAX) [inline ivy/functional/frontends/torch/dtype.py:47]: Same issue with void function here @vedpatwardhan

**Review by KareemMAX — COMMENTED** [review]
LGTM! There are some comments that I'm thinking they should be resolved.

Also @vedpatwardhan, I found some issues with function typehints where they are annotated with `def -> None`, I think that is a mistake. And it causes `ruff` to force the function to `return None`. I'm not really sure if that is intentional, if it was please let me know (I've mentioned you in all of those instances below). If it's not intentional feel free to let me know (no need to rereview the PR, just this part)

**Review by vedpatwardhan — COMMENTED** [review]
> Also @vedpatwardhan, I found some issues with function typehints where they are annotated with `def -> None`, I think that is a mistake. And it causes `ruff` to force the function to `return None`. I'm not really sure if that is intentional, if it was please let me know (I've mentioned you in all of those instances below). If it's not intentional feel free to let me know (no need to rereview the PR, just this part)

Yep those should be void functions, don't think there'll be any issues removing the `-> None`, so feel free to remove it, thanks @KareemMAX @Sai-Suraj-27 😄

File: ivy/functional/backends/numpy/random.py (hunk 1 above)
Comment (Sai-Suraj-27) [reply ivy/functional/backends/numpy/random.py]: @KareemMAX, I have made the changes as per the reply from Ved. Please look into it now. Thank you.

**Review by KareemMAX — APPROVED** [review]

**Pull request merged**